backend/src/main/java/capstone/backend/services/OrderToSupplierService.java

```diff
@@ -45,43 +50,63 @@ public OrderToSupplierDTO createOrder(OrderToSupplierDTO order) throws EntityNot
         return mapOrder(repo.save(mapOrder(order)));
     }
 
-    private boolean checkIfProductsExistent(OrderToSupplierDTO order) {
+    public OrderToSupplierDTO receiveOrder(OrderToSupplierDTO order, OrderStatus status) throws EntityNotFoundException, IllegalArgumentException {
+        if (status != RECEIVED) {
+            throw new IllegalArgumentException("We couldn't process your request!");
+        }
+        if (!orderExists(order)) {
+            throw new EntityNotFoundException("The order you're trying to receive doesn't exist");
+        }
+        if(order.getStatus() == RECEIVED){
+            throw new IllegalArgumentException("The order you're trying to receive has already been received");
```

Comment:
I'm again confused by the "receive" - this is from the supplier's point of view, is it? Maybe rather use "retrieve" or something like that? Maybe this doesn't make a difference after all...

Reply:
you send an order to your supplier. Once you receive the order you click on receive order and the items on the list will be added to your warehous